jsonpb/jsonpb.go

```diff
@@ -128,6 +131,19 @@ type JSONPBUnmarshaler interface {
 
 // Marshal marshals a protocol buffer into JSON.
 func (m *Marshaler) Marshal(out io.Writer, pb proto.Message) error {
+	if msgV2, ok := pb.(protov2.Message); ok {
```

Comment:
what is the thinking behind placing this v2 code path (and early return) before the nil and required field checks below which the v1 code path follows?

Reply:
That stuff except for maybe the nil check would be invalid for v2. Also this is a jsonpb so not performance critical.